lib/auth/password.go

```diff
@@ -134,6 +134,15 @@ func (s *AuthServer) getOTPType(user string) (string, error) {
 	_, err := s.GetHOTP(user)
 	if err != nil {
 		if trace.IsNotFound(err) {
+			_, err := s.GetTOTP(user)
```

Comment:
I don't understand this logic - if notfound, call again? and then return empty?

Reply:
same here. also the level of nesting is higher than we usually keep.

Reply:
Currently this method has a bug as it does not properly detect the OTP Type. 

Since we store OTP and TOTP values in separate files, both reads are required to find the current OTP type. If both are missing then this method should return ''